### PR TITLE
refactor/string: remove from-c-str

### PR DESCRIPTION
**Pull request by m-cat**

### CODE REVIEW

File: src/string.rs [hunk 1]
```diff
@@ -20,12 +20,15 @@ impl ReprC for String {
...
 
-    unsafe fn clone_from_repr_c(c_repr: Self::C) -> Result<String, StringError> {
-        Ok(if c_repr.is_null() {
-            String::default()
-        } else {
-            from_c_str(c_repr)?
-        })
+    unsafe fn clone_from_repr_c(c_repr: Self::C) -> Result<Self, Self::Error> {
+        if c_repr.is_null() {
+            // Return an error instead of an empty String, as a null pointer input is most likely a
```
Comment (m-cat) [inline src/string.rs]: Not sure if we rely on this behavior anywhere currently (null-pointer = empty string). I would prefer not, as a null-pointer in our FFI strategy usually stands for the `None` type of an `Option`. Something to watch for when we start to review this change.

File: src/string.rs [hunk 2]
```diff
@@ -57,14 +60,3 @@ impl From<IntoStringError> for StringError {
         StringError::IntoString(e.description().to_owned())
     }
 }
-
-/// Copies memory from a provided pointer and allocates a new `String`.
-#[inline]
-pub unsafe fn from_c_str(ptr: *const c_char) -> Result<String, StringError> {
```
Comment (nbaksalyar) [inline src/string.rs]: I would not change the API right now. `safe_core` uses this function and this change would require us to change the code in safe_core too, so probably better to leave it for later.

Comment (m-cat) [reply src/string.rs]: I agree that we should wait until we start the FFI changes. However, if we were to do a minor version change after this PR then safe_core can continue using the current version and only upgrade when it's ready to do so.

File: src/string.rs (hunk 1 above)
Comment (nbaksalyar) [reply src/string.rs]: >Not sure if we rely on this behavior anywhere currently (null-pointer = empty string)

As far as I remember, we observed this behaviour with Java/JNI string conversions, so would be worth re-testing the Android builds when we switch to this new API.

**Pull request merged**